src/openms_gui/source/VISUAL/TOPPASScene.cpp

```diff
@@ -480,26 +488,64 @@ namespace OpenMS
 
   bool TOPPASScene::isEdgeAllowed_(TOPPASVertex* u, TOPPASVertex* v)
   {
-    if (u == 0 ||
-        v == 0 ||
-        u == v ||
+    if (u == 0 || v == 0 || u == v ||
         // edges leading to input files make no sense:
         qobject_cast<TOPPASInputFileListVertex*>(v) ||
         // neither do edges coming from output files:
         qobject_cast<TOPPASOutputFileListVertex*>(u) ||
-        // nor edges from input to output without a tool in between:
-        (qobject_cast<TOPPASInputFileListVertex*>(u)
-        && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
-        // nor multiple incoming edges for a single output file/list node
-        (qobject_cast<TOPPASOutputFileListVertex*>(v)
-        && v->inEdgesBegin() != v->inEdgesEnd()) ||
-        // nor mergers connected directly to an output node
-        (qobject_cast<TOPPASMergerVertex*>(u)
-        && qobject_cast<TOPPASOutputFileListVertex*>(v)))
+        // nor edges from input/merger/splitter directly to output:
+        ((qobject_cast<TOPPASInputFileListVertex*>(u) || qobject_cast<TOPPASMergerVertex*>(u) || qobject_cast<TOPPASSplitterVertex*>(u)) && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
+        // nor multiple incoming edges for an output or splitter node:
```

Comment:
With multiple input edges, the splitter also has to do merging (because there's only one output), and there are multiple ways to do the merge. So to avoid confusion, I think your potential use case should be handled by a Merger/Collector followed by a Splitter.